Behavior of non-versioned and versioned cluster tasks (#45917)
openshift/openshift-docs: cicd/pipelines/managing-nonversioned-and-versioned-cluster-tasks.adoc

:_content-type: ASSEMBLY
[id="managing-nonversioned-and-versioned-cluster-tasks"]
= Managing non-versioned and versioned cluster tasks
include::_attributes/common-attributes.adoc[]
:context: managing-nonversioned-and-versioned-cluster-tasks

toc::[]

> [role="_abstract"]

Reviewer: `[role="_abstract"]` is optional. No longer needed but harmless.

When cluster administrators install the {pipelines-title} Operator, two variants of each default cluster tasks are installed -- the non-versioned cluster tasks (NVCT) and the versioned cluster tasks (VCT). For example, the non-versioned `buildah` cluster task and the versioned `buildah-v-1-7` cluster task are installed with the {pipelines-title} Operator v1.7. 

> The metadata, specifications such as `params`, `workspaces`, and `steps`, and the behavior of both variants are same. However, the names are different because the two variants of the same cluster task behave differently during Operator upgrades, and when cluster administrators try to disable cluster tasks. 

Reviewer: ```suggestion
Both variants of a cluster task have the same metadata, behavior, and specifications, including `params`, `workspaces`, and `steps`. However, the variants behave differently during Operator upgrades, and when cluster administrators try to disable them. 
```

Reviewer: "Do. Or do not. There is no try." ~Yoda
